packages/ayzek-core/events/custom.ts

```diff
@@ -13,6 +13,44 @@ export type CustomEventConstructor<I> = {
 	[EVENT_ID]: string,
 	new(...t: any[]): I,
 };
+
+export class TypedEvent<T> {
+	private listeners: Listener<T>[] = [];
+	private listenersOncer: Listener<T>[] = [];
+
+	on(listener: Listener<T>): Disposable {
+		this.listeners.push(listener);
+		return {
+			dispose: () => this.off(listener)
+		};
+	}
+
+	once(listener: Listener<T>): void {
+		this.listenersOncer.push(listener);
+	}
+
+	off(listener: Listener<T>) {
+		var callbackIndex = this.listeners.indexOf(listener);
+		if (callbackIndex > -1) this.listeners.splice(callbackIndex, 1);
+	}
+
+	async emit(event: T) {
+		for (let i = 0; i<this.listeners.length; i++) {
+			await this.listeners[i](event);
+		}
+		if (this.listenersOncer.length > 0) {
+			const toCall = this.listenersOncer;
+			this.listenersOncer = [];
+			for (let i = 0; i<toCall.length; i++) {
```

Comment:
Fix code formatting (Did you ignored vscode extensions recomendations? This project recomends eslint, and such errors should be prohibited by it)